src/winrpath.cxx

```diff
@@ -0,0 +1,1025 @@
+#pragma once
+#include "winrpath.h"
+#include "utils.h"
+
+#include <sstream>
+
+#define BUFSIZE 4096
+
+using CoffMembers = std::vector<coff_entry>;
+
+
+const std::map<char, char> special_character_to_path{
+    {'|', '\\'},
+    {';', ':'}
+};
+
+const std::map<char, char> path_to_special_characters{
+    {'\\', '|'},
+    {'/', '|'},
+    {':', ';'}
+};
+
+/**
+ * Replaces characters used to mangle path characters with
+ * valid path characters
+ * 
+ * \param in a pointer to the string to replace the mangled path characters in
+ * \param len the length of the mangled path
+ */
+void replace_special_characters(char in[], int len)
+{
+    for (int i = 0; i < len; ++i) {
+        if (special_character_to_path.count(in[i]))
+            in[i] = special_character_to_path.at(in[i]);
+    }
+}
+
+/**
+ * Replaces path characters with special, non path, replacement characters
+ * 
+ * \param in a pointer to the string to have its path characters replace with special placeholders
+ * \param len the length of the path to be mangled
+ */
+void replace_path_characters(char in[], int len)
+{
+    for (int i = 0; i < len; i++ ) {
+        if (path_to_special_characters.count(in[i]))
+            in[i] = path_to_special_characters.at(in[i]);
+    }
+}
+
+/**
+ * Pads a given path with an amount of padding of special characters
+ * 
+ * \param pth a pointer to the path to be padded
+ * \param str_size the length of the path
+ * \param bsize the lengh of the padding to add
+ */
+char * pad_path(const char *pth, DWORD str_size, DWORD bsize = MAX_PATH)
+{
+    size_t extended_buf = bsize - str_size;
+    char * padded_path = new char[bsize+1];
+    for(int i = 0, j = 0; i < bsize, j < str_size; i++){
+        if(i < 2){
+            padded_path[i] = pth[j];
+            ++j;
+        }
+        else if(i < extended_buf){
+            padded_path[i] = '|';
+        }
+        else{
+            padded_path[i] = pth[j];
+            ++j;
+        }
+    }
+    return padded_path;
+}
+
+/**
+ * Given a padded library path, return how much the path
+ *  has been padded
+ * 
+ *  \param name the path for which to determine pad count
+ */
+int get_padding_length(const std::string &name)
+{
+    int c = 0;
+    std::string::const_iterator p = name.cbegin();
+    p+=2;
+    while(p != name.end() && *p == '/') {
```

Comment:
(question) I'm missing the corresponding part that inserts these characters.

Reply:
The part that inserts these characters is found in the call to `lib.exe` in the `invokeToolchain` method of the linker class. The actual name is created by the mangling/padding method, which is currently disabled.